### PR TITLE
Add option to opt into full-RBF when sending funds

### PR DESCRIPTION
**Pull request by petertodd**
Easy first-step for those who are using third-party scripts like my [replace-by-fee-tools](https://github.com/petertodd/replace-by-fee-tools)

### CODE REVIEW

**Comment by petertodd** [comment]
There's a good argument for making this default to true - BlockCypher is rejecting Bitcoin Core txs as "might be doublespent", and whether or not the tx fee is high enough to satisfy such services isn't reliable anyway. Defaulting to true also makes it much easier to give people help on fixing a stuck transaction after the fact, even without explicit support in Bitcoin Core yet.

**Comment by TheBlueMatt** [comment]
Your usage docs have a copy paste error... It's a binary flag so you want s/=<amt>//

**Comment by luke-jr** [comment]
This is kinda ugly (per-node start instead of per-tx). IMO we shouldn't opt-in by default until Core can actually double-spend them itself.

**Comment by petertodd** [comment]
@TheBlueMatt Thanks, fixed.

@luke-jr Yeah, per-tx would be nice, but that'll take a lot more work on UX and RPC calls.

Re: default, remember that we don't try to make any guarantees about the likely hood that a tx will be mined according to what the recipient needs re: double-spending. For instance, a low fee tx is very easy to double-spend, and we'll happily produce them. Similarly we'll happily spend unconfirmed change in many circumstances, which again causes txs to get rejected by zeroconf risk analysis's. Not a particularly predictable UX, so i'm not worried about a default to opt-in, which does have benefits. (also the benefit of ensuring opt-in txs are accepted properly by other wallets!)

**Comment by instagibbs** [comment]
The opt-in being standard makes it sound more opt-out from the Core user's perspective. I don't care a ton, but it should to be clear from description. 

I didn't even quite get that:  
>There's a good argument for making this default to true  

meant that's how it was until I read the code itself.

**Comment by sandakersmann** [comment]
NACK. This is not Bitcoin.

**Comment by sipa** [comment]
I don't think we should make this default.

**Comment by TheBlueMatt** [comment]
I agree with Luke, it should be a per-tx option, but this should absolutely be the default. The Bitcoin Core wallet should be setting the example of how to best interact with Bitcoin, which means fill RBF eventually, but normalizing txn with RBF enabled until then.

**Comment by sandakersmann** [comment]
0-confirmation transactions are important. If people include a too small fee they can wait until the transaction drops from the mempool.

**Comment by rnicoll** [comment]
Having the option is important, but I'd much rather it be per-transaction, and I'm certainly not convinced RBF-enabled should be default. What's the argument for it as default, that it's impossible to be certain of required transaction fee before relaying?

Maybe the default for fee-less transactions only? Aware that's somewhat complicating things.

**Comment by TheBlueMatt** [comment]
The best way to optimize fee for users is to always use RBF and shoot for a low fee initially. Certainly this is the way we'd go in the future, the question is whether to default txn to opt into RBF before we've implemented the actual fee-bump-RBF logic in wallet or not.

**Comment by sandakersmann** [comment]
Merchants are users too. RBF are not the way to go in the future.

**Comment by maaku** [comment]
Concept ACK. Branch doesn't currently merge.

**Comment by btcdrak** [comment]
Needs rebase @petertodd

**Comment by surg0r** [comment]
No.

**Comment by gmaxwell** [comment]
Closing for now.

File: src/wallet/wallet.h [hunk 1]
```diff
@@ -49,6 +50,8 @@ static const CAmount DEFAULT_TRANSACTION_MAXFEE = 0.1 * COIN;
 static const CAmount MIN_CHANGE = CENT;
 //! -txconfirmtarget default
 static const unsigned int DEFAULT_TX_CONFIRM_TARGET = 2;
+//! -optintofullrbf default
+static const bool DEFAULT_OPT_INTO_FULL_RBF = true;
```
Comment (MarcoFalke) [inline src/wallet/wallet.h]: Could this make sense as a separate PR with `DEFAULT_OPT_INTO_FULL_RBF = false`?

Comment (btcdrak) [inline src/wallet/wallet.h]: Yes. This PR is uncontroversial if this value defaults to false since we're just adding command line configuration options.
